### ISSUE TITLE
Network Manager not available

### ISSUE THREAD

**Issue by lindemann09** [eduvpn/python-eduvpn-client#391]
# Installation

How did you install the eduVPN client?
```
sudo apt install eduvpn-client
```

# Version

What version of the client you are running? Try your package manager or otherwise `$ eduvpn-client -v`.

```
➜  ~ dpkg -l | grep eduvpn
ii  eduvpn-client                         2.0.0-1                               all          VPN client for educational networks
ii  eduvpn-client-data                    2.0.0-1                               all          VPN client for educational networks (data files)
ii  python3-eduvpn-client                 2.0.0-1                               all          eduVPN client (Python 3)
```

# OS/Distribution

What operating system/distribution and version you are running?
```
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.2 LTS
Release:	20.04
Codename:	focal
```

new installation of Ubuntu 20.04

# Logs

## Do you have a problem while adding a profile?

Eduvpn will print out information to the console while running. Please try running `eduvpn-client` in a console.  Are there any interesting logs appearing when you try to connect? There is also a more verbose mode available which is enabled with the `-d` flag.

```
➜  ~ eduvpn-cli interactive   
**Network Manager not available**
INFO:eduvpn.remote:Requesting https://disco.eduvpn.org/v2/server_list.json
INFO:eduvpn.remote:Retrieving signature https://disco.eduvpn.org/v2/server_list.json.minisig
INFO:eduvpn.remote:verifying signature of https://disco.eduvpn.org/v2/server_list.json
INFO:eduvpn.remote:Requesting https://disco.eduvpn.org/v2/organization_list.json
INFO:eduvpn.remote:Retrieving signature https://disco.eduvpn.org/v2/organization_list.json.minisig
INFO:eduvpn.remote:verifying signature of https://disco.eduvpn.org/v2/organization_list.json

Please choose server:

.....

> 15
Traceback (most recent call last):
  File "/usr/bin/eduvpn-cli", line 11, in <module>
    load_entry_point('eduvpn-client==2.0.0', 'console_scripts', 'eduvpn-cli')()
  File "/usr/lib/python3/dist-packages/eduvpn/cli.py", line 139, in eduvpn
    parse_eduvpn(argv[1:])
  File "/usr/lib/python3/dist-packages/eduvpn/cli.py", line 105, in parse_eduvpn
    parsed.func(parsed)
  File "/usr/lib/python3/dist-packages/eduvpn/cli.py", line 66, in interactive
    auth_url, display_name, support_contact, secure_internets = menu.interactive(args)
  File "/usr/lib/python3/dist-packages/eduvpn/menu.py", line 182, in interactive
    choice = menu(institutes=institute_access, orgs=orgs, search_term=search_term)
  File "/usr/lib/python3/dist-packages/eduvpn/menu.py", line 75, in menu
    return provider_choice(institutes, orgs)
  File "/usr/lib/python3/dist-packages/eduvpn/menu.py", line 58, in provider_choice
    return institute['base_url'], extract_translation(institute['display_name']), institute[
KeyError: 'support_contact'
```
## eduvpn-gui does also not work

```
➜  ~ eduvpn-gui 
2021-07-07 10:57:27,119 - MainThread - ERROR - eduvpn.ui.__main__ - __main__.py:77 - Caught exception: Namespace NM not available
2021-07-07 10:57:27,119 - MainThread - ERROR - root - __main__.py:78 - Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/eduvpn/ui/__main__.py", line 71, in main_loop
    from eduvpn.ui.ui import EduVpnGui
  File "/usr/lib/python3/dist-packages/eduvpn/ui/ui.py", line 15, in <module>
    gi.require_version('NM', '1.0')  # noqa: E402
  File "/usr/lib/python3/dist-packages/gi/__init__.py", line 129, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace NM not available
```

## Do you have a problem during connecting?

Please examine the eduvpn for errors or messages while connecting. Note that the actual VPN connection management is not done by eduVPN, but by network management tool bundeled with your OS named networkManager. You can examine the NetworkManager logs with `$ sudo journalctl -u NetworkManager`.

```
Jul 07 11:05:03 oliver-desktop systemd[1]: Starting Network Manager...
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.8141] NetworkManager (version 1.22.10) is starting... (for the first time)
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.8142] Read config: /etc/NetworkManager/NetworkManager.conf (lib: 10-dns-resolved.conf, no-mac-addr-change.conf) (run: 10-g>
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.8709] bus-manager: acquired D-Bus service "org.freedesktop.NetworkManager"
Jul 07 11:05:03 oliver-desktop systemd[1]: Started Network Manager.
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.8755] manager[0x55d6a4790030]: monitoring kernel firmware directory '/lib/firmware'.
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.8755] monitoring ifupdown state file '/run/network/ifstate'.
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9591] hostname: hostname: using hostnamed
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9591] hostname: hostname changed from (none) to "oliver-desktop"
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9597] dns-mgr[0x55d6a4778290]: init: dns=systemd-resolved rc-manager=symlink, plugin=systemd-resolved
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9603] manager[0x55d6a4790030]: rfkill: Wi-Fi hardware radio set enabled
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9603] manager[0x55d6a4790030]: rfkill: WWAN hardware radio set enabled
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9642] Loaded device plugin: NMAtmManager (/usr/lib/x86_64-linux-gnu/NetworkManager/1.22.10/libnm-device-plugin-adsl.so)
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9656] Loaded device plugin: NMTeamFactory (/usr/lib/x86_64-linux-gnu/NetworkManager/1.22.10/libnm-device-plugin-team.so)
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9671] Loaded device plugin: NMWwanFactory (/usr/lib/x86_64-linux-gnu/NetworkManager/1.22.10/libnm-device-plugin-wwan.so)
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9679] Loaded device plugin: NMWifiFactory (/usr/lib/x86_64-linux-gnu/NetworkManager/1.22.10/libnm-device-plugin-wifi.so)
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9690] Loaded device plugin: NMBluezManager (/usr/lib/x86_64-linux-gnu/NetworkManager/1.22.10/libnm-device-plugin-bluetooth>
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9693] manager: rfkill: Wi-Fi enabled by radio killswitch; enabled by state file
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9694] manager: rfkill: WWAN enabled by radio killswitch; enabled by state file
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9695] manager: Networking is enabled by state file
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9696] dhcp-init: Using DHCP client 'internal'
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9700] settings: Loaded settings plugin: ifupdown ("/usr/lib/x86_64-linux-gnu/NetworkManager/1.22.10/libnm-settings-plugin->
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9701] settings: Loaded settings plugin: keyfile (internal)
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9701] ifupdown: management mode: unmanaged
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9701] ifupdown:       interface-parser: parsing file /etc/network/interfaces
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9701] ifupdown:       interface-parser: source line includes interfaces file(s) /etc/network/interfaces.d
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9701] ifupdown:       interface-parser: finished parsing file /etc/network/interfaces
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9740] device (lo): carrier: link connected
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9745] manager: (lo): new Generic device (/org/freedesktop/NetworkManager/Devices/1)
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9758] manager: (enp7s0): new Ethernet device (/org/freedesktop/NetworkManager/Devices/2)
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9778] settings: (enp7s0): created default wired connection 'Wired connection 1'
Jul 07 11:05:03 oliver-desktop NetworkManager[1150]: <info>  [1625648703.9784] device (enp7s0): state change: unmanaged -> unavailable (reason 'managed', sys-iface-state: 'external')
Jul 07 11:05:04 oliver-desktop NetworkManager[1150]: <info>  [1625648704.0149] modem-manager: ModemManager available
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3486] device (enp7s0): carrier: link connected
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3493] device (enp7s0): state change: unavailable -> disconnected (reason 'carrier-changed', sys-iface-state: 'managed')
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3507] policy: auto-activating connection 'Wired connection 1' (6ecfb129-f239-3a13-8743-2b8046776e3d)
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3516] device (enp7s0): Activation: starting connection 'Wired connection 1' (6ecfb129-f239-3a13-8743-2b8046776e3d)
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3518] device (enp7s0): state change: disconnected -> prepare (reason 'none', sys-iface-state: 'managed')
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3524] manager: NetworkManager state is now CONNECTING
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3528] device (enp7s0): state change: prepare -> config (reason 'none', sys-iface-state: 'managed')
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3538] device (enp7s0): state change: config -> ip-config (reason 'none', sys-iface-state: 'managed')
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3544] dhcp4 (enp7s0): activation: beginning transaction (timeout in 45 seconds)
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3664] dhcp4 (enp7s0): option dhcp_lease_time      => '864000'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3664] dhcp4 (enp7s0): option domain_name          => 'fritz.box'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3664] dhcp4 (enp7s0): option domain_name_servers  => '192.168.178.1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option expiry               => '1626512709'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option ip_address           => '192.168.178.3'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option next_server          => '192.168.178.1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option ntp_servers          => '192.168.178.1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option requested_broadcast_address => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option requested_domain_name => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option requested_domain_name_servers => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option requested_domain_search => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option requested_host_name  => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option requested_interface_mtu => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option requested_ms_classless_static_routes => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option requested_nis_domain => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option requested_nis_servers => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3665] dhcp4 (enp7s0): option requested_ntp_servers => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3666] dhcp4 (enp7s0): option requested_rfc3442_classless_static_routes => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3666] dhcp4 (enp7s0): option requested_root_path  => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3666] dhcp4 (enp7s0): option requested_routers    => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3666] dhcp4 (enp7s0): option requested_static_routes => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3666] dhcp4 (enp7s0): option requested_subnet_mask => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3666] dhcp4 (enp7s0): option requested_time_offset => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3666] dhcp4 (enp7s0): option requested_wpad       => '1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3666] dhcp4 (enp7s0): option routers              => '192.168.178.1'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3666] dhcp4 (enp7s0): option subnet_mask          => '255.255.255.0'
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3666] dhcp4 (enp7s0): state changed unknown -> bound
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3687] device (enp7s0): state change: ip-config -> ip-check (reason 'none', sys-iface-state: 'managed')
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3718] device (enp7s0): state change: ip-check -> secondaries (reason 'none', sys-iface-state: 'managed')
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3723] device (enp7s0): state change: secondaries -> activated (reason 'none', sys-iface-state: 'managed')
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3731] manager: NetworkManager state is now CONNECTED_LOCAL
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3750] manager: NetworkManager state is now CONNECTED_SITE
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3752] policy: set 'Wired connection 1' (enp7s0) as default for IPv4 routing and DNS
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3760] device (enp7s0): Activation: successful, device activated.
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3765] manager: NetworkManager state is now CONNECTED_GLOBAL
Jul 07 11:05:09 oliver-desktop NetworkManager[1150]: <info>  [1625648709.3769] manager: startup complete
Jul 07 11:05:11 oliver-desktop NetworkManager[1150]: <info>  [1625648711.1482] dhcp6 (enp7s0): activation: beginning transaction (timeout in 45 seconds)
Jul 07 11:05:11 oliver-desktop NetworkManager[1150]: <info>  [1625648711.1496] policy: set 'Wired connection 1' (enp7s0) as default for IPv6 routing and DNS
Jul 07 11:05:11 oliver-desktop NetworkManager[1150]: <info>  [1625648711.1537] dhcp6 (enp7s0): option dhcp6_name_servers   => 'fd00::9a9b:cbff:fe50:f4b0'
Jul 07 11:05:11 oliver-desktop NetworkManager[1150]: <info>  [1625648711.1538] dhcp6 (enp7s0): state changed unknown -> bound
Jul 07 11:05:14 oliver-desktop NetworkManager[1150]: <info>  [1625648714.3955] agent-manager: agent[c50bb1f690129d59,:1.77/org.freedesktop.nm-applet/1000]: agent registered
```

**Comment by lindemann09** [comment]
Some Ubuntu flavors require the package  `gir1.2-nm-1.0`. This should documented.

```
sudo apt install gir1.2-nm-1.0
```

**Comment by gijzelaerr** [comment]
This is actually a duplicate of https://github.com/eduvpn/python-eduvpn-client/issues/358, which I just fixed for 2.0.0-2 and pushed to the deb repo. let me know if this solves your issue or causes new problems.
